web/src/lib/data-hooks/useUserData.ts

```diff
@@ -53,10 +55,22 @@ export const useUserData = (): UseUserDataResponse => {
 
   useEffect(() => {
     fetchedUserId.current = data?.user.id;
+
+    let licenseDataFromDatabaseDataMoreRecent = false;
+    if (data?.businesses[data.currentBusinessId] && updateQueue?.currentBusiness()) {
+      licenseDataFromDatabaseDataMoreRecent = isLicenseDataFromDatabaseDataMoreRecent({
+        businessFromDb: data?.businesses[data.currentBusinessId],
+        businessFromUpdateQueue: updateQueue?.currentBusiness(),
+      });
+    }
+
     if (updateQueue === undefined && data) {
       setUpdateQueue(new UpdateQueueFactory(data, update));
     } else if (updateQueue?.current() === undefined && data) {
       updateQueue?.queue(data);
+    } else if (licenseDataFromDatabaseDataMoreRecent && updateQueue && data) {
```

Comment:
I'm not sure how to test this, also I noticed there are no tests for lines 67-70. I'm guessing it was intentional. Perhaps we can't create them using the existing approach.